components/IdentifyViewer.jsx

```diff
@@ -152,6 +152,74 @@ const BuiltinExporters = [
                 });
             }
         }
+    }, {
+        id: 'shapefile',
+        title: 'Shapefile',
+        allowClipboard: false,
+        export: (json, callback) => {
+            import("@mapbox/shp-write").then(shpwriteMod => {
+                const shpwrite = shpwriteMod.default;
+                const layers = Object.entries(json);
+                const options = {
+                    outputType: 'arraybuffer',
+                    types: {
+                        point: 'points',
+                        polygon: 'polygons',
+                        polyline: 'lines'
+                    }
+                };
+                if (layers.length === 1) {
```

Comment:
To reduce code duplication, you could directly do

    const promises = layers.map(([layerName, features]) => { .... });

and then at the end, depending on the number of layers, decide whether to zip the result or return the single shp zip directly.